File: api/webhook.js
const BOT_TOKEN = process.env.BOT_TOKEN;
const CHAT_ID = process.env.CHAT_ID;

export default async function handler(req, res) {
  if (req.method !== 'POST') return res.status(200).json({ ok: true });

  try {
    const { message } = req.body;
    if (!message) return res.status(200).json({ ok: true });

    const text = message.text || '';
    const from = message.from;
    const userName = from.first_name + (from.last_name ? ' ' + from.last_name : '');
    const userHandle = from.username ? '@' + from.username : 'нет username';

    let reply = '';

    if (text === '/start') {
      reply = `Привет! 👋\n\nЭто бот Глеба Дохевича — персонального фитнес тренера.\n\n💪 Напиши мне напрямую:\n✈ Telegram: @DoHevich\n📞 Телефон: +7 (965) 948-69-24\n\nИли оставь заявку на сайте — Глеб ответит в течение часа!`;
    } else {
      reply = `Спасибо за сообщение! Глеб скоро свяжется с тобой. 💪\n\nИли напиши напрямую: @DoHevich`;
    }

    // Send reply to user
    await fetch(`https://api.telegram.org/bot${BOT_TOKEN}/sendMessage`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ chat_id: message.chat.id, text: reply })
    });

    res.status(200).json({ ok: true });
  } catch (e) {
    console.error(e);
    res.status(200).json({ ok: true });
  }
}

File: api/webhook.py
import json
import os
import urllib.request

BOT_TOKEN = os.environ.get("BOT_TOKEN", "")
CHAT_ID   = os.environ.get("CHAT_ID", "")

def send_message(text):
    url  = f"https://api.telegram.org/bot{BOT_TOKEN}/sendMessage"
    data = json.dumps({
        "chat_id":    CHAT_ID,
        "text":       text,
        "parse_mode": "HTML"
    }).encode("utf-8")
    req = urllib.request.Request(
        url, data=data,
        headers={"Content-Type": "application/json"}
    )
    urllib.request.urlopen(req)

def handler(request):
    # CORS preflight
    if request.method == "OPTIONS":
        return {
            "statusCode": 200,
            "headers": {
                "Access-Control-Allow-Origin":  "*",
                "Access-Control-Allow-Methods": "POST, OPTIONS",
                "Access-Control-Allow-Headers": "Content-Type"
            },
            "body": ""
        }

    if request.method == "POST":
        try:
            body = request.body
            if isinstance(body, bytes):
                body = body.decode("utf-8")
            data = json.loads(body)

            name    = data.get("name",    "—")
            phone   = data.get("phone",   "—")
            goal    = data.get("goal",    "—")
            comment = data.get("comment", "—")

            msg = (
                f"🔥 <b>НОВАЯ ЗАЯВКА С САЙТА!</b>\n\n"
                f"👤 <b>Имя:</b> {name}\n"
                f"📱 <b>Контакт:</b> {phone}\n"
                f"🎯 <b>Цель:</b> {goal}\n"
                f"💬 <b>Комментарий:</b> {comment}\n\n"
                f"⏱ Ответь клиенту как можно скорее!"
            )
            send_message(msg)

            return {
                "statusCode": 200,
                "headers": {
                    "Access-Control-Allow-Origin": "*",
                    "Content-Type": "application/json"
                },
                "body": json.dumps({"ok": True})
            }

        except Exception as e:
            return {
                "statusCode": 500,
                "headers": {"Access-Control-Allow-Origin": "*"},
                "body": json.dumps({"error": str(e)})
            }

    return {
        "statusCode": 200,
        "headers": {"Access-Control-Allow-Origin": "*"},
        "body": json.dumps({"ok": True})
    }
